src/users/containers/edit-user-permissions/edit-user-permission.component.ts

```diff
@@ -0,0 +1,116 @@
+    selector: 'edit-user-permission',
+    templateUrl: './edit-user-permission.component.html',
+  })
+  export class EditUserPermissionComponent  implements OnInit, OnDestroy {
+    editUserForm: FormGroup;
+    isInvalid;
+    errorMessages = {
+      roles: ['Select at least one option'],
+    };
+    user$: Observable<any>;
+    isLoading$: Observable<boolean>;
+    user: any;
+    isPuiCaseManager: boolean;
+    isPuiOrganisationManager: boolean;
+    isPuiUserManager: boolean;
+    userId: string;
+
+    userSubscription: Subscription;
+    dependanciesSubscription: Subscription;
+    backUrl: string;
+
+    constructor(
+      private userStore: Store<fromStore.UserState>,
+      private routerStore: Store<fromRoot.State>
+    ) { }
+
+    ngOnInit(): void {
+      this.isLoading$ = this.userStore.pipe(select(fromStore.getGetUserLoading));
+
+      this.dependanciesSubscription = combineLatest([
+      this.routerStore.pipe(select(fromRoot.getRouterState)),
+      this.userStore.pipe(select(fromStore.getGetUserLoaded))
+    ]).subscribe(([route, users]) => {
+      if (users === false) {
+        this.userStore.dispatch(new fromStore.LoadUsers());
+      }
+      this.userId = route.state.params.userId;
+      this.user$ = this.userStore.pipe(select(fromStore.getGetSingleUser, { userIdentifier: this.userId }));
+      this.backUrl = this.getBackurl(this.userId);
+    });
+
+      this.userSubscription = this.user$.subscribe((user) => {
+        this.user = user;
+        this.isPuiCaseManager = this.getIsPuiCaseManager(user);
+        this.isPuiOrganisationManager = this.getIsPuiOrganisationManager(user);
+        this.isPuiUserManager = this.getIsPuiUserManager(user);
+
+        this.editUserForm = this.getFormGroup(this.isPuiCaseManager,
+          this.isPuiUserManager, this.isPuiOrganisationManager, checkboxesBeCheckedValidator);
+      });
+    }
+
+  getBackurl(userId: string) {
+    return `/users/user/${userId}`;
+  }
+
+  getFormGroup(isPuiCaseManager, isPuiUserManager, isPuiOrganisationManager, validator: any): FormGroup {
+    return new FormGroup({
+      roles: new FormGroup({
+        'pui-case-manager': new FormControl(isPuiCaseManager),
+        'pui-user-manager': new FormControl(isPuiUserManager),
+        'pui-organisation-manager': new FormControl(isPuiOrganisationManager)
+      }, validator(1))
+    });
+  }
+
+  getIsPuiCaseManager(user: any): boolean {
+    return user && user.manageCases === 'Yes';
+  }
+
+  getIsPuiOrganisationManager(user: any): boolean {
+    return user && user.manageOrganisations === 'Yes';
+  }
+
+  getIsPuiUserManager(user: any): boolean {
+    return user && user.manageUsers === 'Yes';
+  }
+
+  unsubscribe(subscription: Subscription) {
+    if (subscription) {
+      subscription.unsubscribe();
+    }
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe(this.userSubscription);
+    this.unsubscribe(this.dependanciesSubscription);
+  }
+
+  onSubmit() {
+    if (this.editUserForm.valid) {
```

Comment:
Think about using a JS early return here, to get rid of the nesting.

Reply:
Done